### PR TITLE
proxychains-ng: init at 4.15

### PR DESCRIPTION
**Pull request by ZenithalHourlyRate**
###### Motivation for this change

Add proxychains-ng which is more maintained than proxychains.

Ref to #67688, which is stale.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -12492,4 +12492,14 @@
     github = "rski";
     githubId = 2960312;
   };
+  zenithal = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:12495]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (ZenithalHourlyRate) [reply maintainers/maintainer-list.nix]: Done.

It seems that `maintainers/maintainer-list.nix` is not sorted, so I inserted myself in the proper place and left others alone.

**Comment by bobby285271** [comment]
Result of `nixpkgs-review pr 137541` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>proxychains-ng</li>
  </ul>
</details>

File: pkgs/tools/networking/proxychains-ng/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    rev = "v${version}";
+    sha256 = "128d502y8pn7q2ls6glx9bvibwzfh321sah5r5li6b6iywh2zqlc";
+  };
+
+  meta = with lib; {
+    description = "A preloader which hooks calls to sockets in dynamically linked programs and redirects it through one or more socks/http proxies";
+    homepage = "https://github.com/rofl0r/proxychains-ng";
+    license = licenses.gpl2Plus;
+    maintainers = with maintainers; [ zenithal ];
+    platforms = platforms.linux;
```
Comment (bobby285271) [inline pkgs/tools/networking/proxychains-ng/default.nix:22]: Is it possible to change this to `platforms.unix`?

**Review by bobby285271 — APPROVED** [review]

File: pkgs/tools/networking/proxychains-ng/default.nix (hunk 2 above)
Comment (ZenithalHourlyRate) [reply pkgs/tools/networking/proxychains-ng/default.nix:22]: According to the README of https://github.com/rofl0r/proxychains-ng, it is a unix program and it supports Linux, BSD, Mac, Haiku.

I would like to make this change, but I could not test it.

Comment (bobby285271) [reply pkgs/tools/networking/proxychains-ng/default.nix:22]: Maybe just check results from ofborg.

Comment (ZenithalHourlyRate) [reply pkgs/tools/networking/proxychains-ng/default.nix]: All checks have passed

Comment (bobby285271) [reply pkgs/tools/networking/proxychains-ng/default.nix]: [It is falling on darwin](https://logs.nix.ci/?key=nixos/nixpkgs.137541&attempt_id=84056fd3-fec8-4c13-98a0-84eaa683459f) (sorry we are lack of darwin reviewers and CI checks for darwin take long time).

Comment (ZenithalHourlyRate) [reply pkgs/tools/networking/proxychains-ng/default.nix]: Then I revert back to `platforms.linux`

**Comment by bobby285271** [comment]
@ofborg eval
